### PR TITLE
fix: update JioTV API compatibility for v7.1.8

### PR DESCRIPTION
**Pull request by harshvasudeva**
## Summary

JioTV Go is currently at **v3.21.0**. This PR updates the upstream API integration for the current JioTV Android app **v7.1.8 (version code 422)**.

## What changed

- Updated the JioTV client and authentication/DRM/logout headers to version code 422.
- Updated the OkHttp/PlayTV user agents and the JioTV API key extracted from the v7.1.8 app.
- Moved channel-list requests to the v3.1 endpoints and updated premium provider API hostnames.
- Re-resolved premium entitlement IDs (for example, `Z0177`) to the content-provider IDs required by catalog/playback APIs.
- Fixed a Web EPG panic for channel IDs shorter than the `sl` prefix and added regression coverage.
- Ignored the local decompiled APK directory so reverse-engineering artifacts are not committed.

## Why

The previous integration targeted the older v7.1.7-era headers and endpoints. JioTV now rejects or misroutes requests using stale app metadata, and premium plans can expose entitlement IDs that the catalog and playback endpoints do not accept directly.

## Validation

- `go build ./...` passes.
- Focused tests pass for `internal/handlers`, `pkg/television`, and `internal/constants/urls`.
- The full `go test ./...` run is currently affected by pre-existing Windows/environment-sensitive failures in `cmd`, `internal/config`, `pkg/epg`, and `pkg/utils` (temporary-file locking, Windows path-separator expectations, and an existing numeric epoch parsing test).

### CODE REVIEW

**Comment by harshvasudeva** [comment]
Root cause confirmed by comparing the official JioTV Android APK v7.1.8 (version code 422) with live requests. The previous implementation used the raw GCP origin hosts `origin-gcp-api-jiotv.media.jio.com` and `origin-gcp-content-jiotv.media.jio.com`. Those hosts now return a generic HTML 403 before authentication is evaluated, indicating a Jio-side cloud/origin firewall or access-policy change.

The v7.1.8 APK no longer references those hosts and uses `api-jiotv.media.jio.com` and `content-jiotv.media.jio.com` instead. After updating to those hosts, the FanCode directory resolves `Z0177` to content provider ID `200169`, and the previously failing catalog route returns HTTP 200 with content.

This is separate from entitlement failures: the test account has FanCode only, so other providers correctly return Jio's “No eligible plans found” response.

**Review by rabilrbl — COMMENTED** [review]
Thanks for the PR @harshvasudeva 

Awesome work, I'll test myself and merge when I get time.

Are you available on Telegram? I wanted discuss few things with you.

**Comment by harshvasudeva** [comment]
No I dont use telegram.

File: .gitignore [hunk 1]
```diff
@@ -59,4 +59,6 @@ jiotv.json
 
 # MdBook Docs
 docs/book/
-book/
\ No newline at end of file
+book/
+# Decompiled APK for reverse-engineering (do not commit)
+apk/
```
Comment (rabilrbl) [inline .gitignore]: Please remove this and practice maintaining the directory outside the repo scope.

**Review by rabilrbl — CHANGES_REQUESTED** [review]
@harshvasudeva Can you please solve merge conflicts?

And 1 change below

**Comment by harshvasudeva** [comment]
Thanks for the review!

**Merge conflicts** — resolved, merged `develop` in. Only 4 files actually overlapped:
- `.gitignore` — kept your `pnpm*` addition, see below on the `apk/` line.
- `internal/handlers/drm.go` — auto-merged cleanly (my one-line versionCode change and your new DASH-clock code sit in different parts of the file).
- `internal/handlers/epg.go` / `epg_test.go` — you'd independently fixed the same `channelID[:2]` panic I had (via `HasPrefix` instead of `TrimPrefix`, as part of the day-offset-correction work). Took your version wholesale rather than keeping two fixes for one bug, and ported my short-channel-ID regression test into your test file — it covers an edge case (`channelID` shorter than 2 chars) your new test suite doesn't hit.

**`apk/` in `.gitignore`** — fair point, done. Removed the folder from the repo working directory entirely rather than just gitignoring it, so there's nothing left to ignore.

All tests still pass (confirmed against a clean checkout of your latest `develop` too, so I'm not counting failures that were already there).

**Review by rabilrbl — APPROVED** [review]
